overlord/devicestate/devicemgr.go

```diff
@@ -891,7 +908,7 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 		}
 		if currentSys != nil && isCurrentSystem(currentSys, system) {
```

Comment:
I still feel this should actually be  integrated back in systemFromSeed by passing all the needed info in, but can be done when moving it to systems.go

Reply:
As discussed, we'll do this after #8689 lands.